modelskill/model/grid.py

```diff
@@ -163,7 +163,12 @@ def _extract_point(
         assert isinstance(self.data, xr.Dataset)
 
         # TODO: avoid runtrip to pandas if possible (potential loss of metadata)
-        ds = self.data.interp(coords=dict(x=x, y=y), method=method)  # type: ignore
+        if "z" in self.data.dims and z is not None:
```

Comment:
Is this better 🤔? 

    coords = dict(x=x, y=y, z=z) if ("z" in self.data.dims and z is not None) else dict(x=x, y=y)
    ds = self.data.interp(coords=coords, method=method)